src/layers/base-layer.ts

```diff
@@ -178,13 +176,14 @@ function* generateColor(): Generator<RGBColor> {
 
 export const colorMaker = generateColor();
 
-class Layer<LayerConfig extends LayerBaseConfig = LayerBaseConfig> {
+class Layer {
   id: string;
   // TODO: define meta
   meta: {};
-  // TODO: define visConfigSettings
-  visConfigSettings: {};
-  config: LayerConfig;
+  visConfigSettings: {
+    [key in keyof Partial<LayerVisConfig>]: ValueOf<LayerVisConfigSettings>;
```

Comment:
Can we just put `[key: string]` here? people can create custom layers and assign custom visConfig props that is not part of the standard `LayerVisConfig`. This should be allowed by BaseLayer